tracing: Change tracing state dynamically at runtime

devbhavej opened the pull request:
This commit adds support for applications to enable/disable tracing dynamically at runtime. Current implementation only allow changing tracing state when `CONFIG_TRACING_HANDLE_HOST_CMD` is enabled. Thus application cannot enable/disable tracing to save tracing buffer after some unexpected event.

devbhavej committed:
tracing: Change tracing state dynamically at runtime

This commit adds support for applications to enable/disable tracing
dynamically at runtime. Currentl implementation only allow changing tracing
state when `CONFIG_TRACING_HANDLE_HOST_CMD` is enabled. Thus application
cannot enable/disable tracing to save tracing buffer after some unexpected
event.

Signed-off-by: Dev Joshi <devbhave@qti.qualcomm.com>

devbhavej commented:
Thanks @teburd for your approval. 

Can anyone help with why these 3 checks are failing? I think they are failing due to some setup issue?

kartben commented on include/zephyr/tracing/tracing.h:
```diff
@@ -2817,6 +2817,12 @@ void sys_trace_idle_exit(void);
+ * @brief Enable/Disable Tracing
+ * @param state A boolean value either true or false
+ */
+#define sys_trace_set_state(state)
```
Please add to the release notes

devbhavej committed:
doc: tracing: Add new API for v4.5 release notes

Add newly added macro/API to set tracing state (i.e. enable/disable)
from the application code base.

Signed-off-by: Dev Joshi <devbhave@qti.qualcomm.com>

teburd reviewed (commented):
I like the feature I think but would like to see some documentation and samples showing how to use it as part of the PR

devbhavej commented:
@teburd Thanks for your comment. Please let me know if you think we need more information around the usage of this API/feature. Thanks!
